app/src/main/java/com/example/bottomnav/Parse.java

```diff
@@ -63,7 +63,7 @@ private void storeConst(CPPASTSimpleDeclaration declaration) throws Exception {
     }
 
     private void storeStuct(IASTDeclaration[] declarations, String name) throws Exception {
-        ArrayList<StructContents> struct = new ArrayList();
+        ArrayList<StructContents> struct = new ArrayList<>();
```

Comment:
Can you get away with this? since you already define the type parameter in the type declaration
```suggestion
        ArrayList<StructContents> struct = new ArrayList);
```